pkg/azuremachine/mutate_create.go

```diff
@@ -83,6 +84,19 @@ func (m *CreateMutator) Mutate(ctx context.Context, request *v1beta1.AdmissionRe
 		result = append(result, *patch)
 	}
 
+	azureMachineCR.Default()
+	{
+		var capiPatches []mutator.PatchOperation
+		capiPatches, err = patches.GeneratePatchesFrom(request.Object.Raw, azureMachineCR)
+		if err != nil {
+			return []mutator.PatchOperation{}, microerror.Mask(err)
+		}
+
+		capiPatches = patches.SkipPatchesForPath("/spec/sshPublicKey", capiPatches)
```

Comment:
We require users to not set this value. We have validation to make sure it's empty, while CAPZ defaults it to a generic key.